core/src/main/java/pl/agh/edu/ui/component/button/ScenarioButton.java

```diff
@@ -0,0 +1,148 @@
+import pl.agh.edu.engine.hotel.HotelType;
+import pl.agh.edu.ui.GameSkin;
+import pl.agh.edu.ui.component.label.LanguageLabel;
+import pl.agh.edu.ui.language.LanguageManager;
+import pl.agh.edu.ui.resolution.ResolutionManager;
+import pl.agh.edu.ui.utils.FontType;
+import pl.agh.edu.ui.utils.SkinColor;
+import pl.agh.edu.ui.utils.wrapper.WrapperContainer;
+
+public class ScenarioButton extends WrapperContainer<Button> {
+	private final Skin skin = GameSkin.getInstance();
+	private final Button button;
+	private final ScenarioButtonStyle scenarioButtonStyle;
+	private int width;
+	private int height;
+	private LanguageLabel titleLabel;
+	private Image scenarioImage;
+	private LanguageLabel descriptionLabel;
+	private LanguageLabel seasonLabel;
+
+	public ScenarioButton(HotelType hotelType) {
+		this.button = new Button(skin.get("scenario-difficulty-button", Button.ButtonStyle.class));
+		this.scenarioButtonStyle = new ScenarioButtonStyle(hotelType);
+		setActor(button);
+		setSize();
+
+		ScenarioButtonStyle.createPad(this);
+		ResolutionManager.addListener(this);
+		LanguageManager.addListener(this);
+
+		createActors();
+		createButton();
+		setTouchable(Touchable.enabled);
+	}
+
+	public void createActors() {
+		titleLabel = createLabel(scenarioButtonStyle.getTitleLabelPath(), scenarioButtonStyle.getTitleLabelFont(), height / 30);
+		titleLabel.setAlignment(getAlign());
+
+		scenarioImage = new Image(skin.getDrawable(scenarioButtonStyle.getIconPath()));
+
+		descriptionLabel = createLabel(scenarioButtonStyle.getDescriptionLabelPath(), scenarioButtonStyle.getDescriptionLabelFont(), height / 40);
+		descriptionLabel.setAlignment(getAlign());
+
+		seasonLabel = createLabel(scenarioButtonStyle.getSeasonLabelPath(), scenarioButtonStyle.getDescriptionLabelFont(), height / 40);
+		seasonLabel.setAlignment(getAlign());
+	}
+
+	public void createButton() {
+		button.clearChildren();
+		button.add(titleLabel).width(5 * width / 24).height(height / 12).row();
+		button.add(scenarioImage).width(width / 12).height(height / 8).padTop(height / 50).center().row();
+		button.add(descriptionLabel).width(5 * width / 24).height(height / 12).padTop(height / 50).row();
+		button.add(seasonLabel).width(3 * width / 24).padTop(height / 40).center();
+	}
+
+	public void setSize() {
+		width = GraphicConfig.getResolution().WIDTH;
+		height = GraphicConfig.getResolution().HEIGHT;
+	}
+
+	private LanguageLabel createLabel(String labelTextPath, String labelFont, int lineHeight) {
+		System.out.println(labelTextPath);
+		Label.LabelStyle labelStyle = skin.get(labelFont, Label.LabelStyle.class);
+		labelStyle.font.getData().setLineHeight(lineHeight);
+		labelStyle.fontColor = SkinColor.ALERT.getColor(SkinColor.ColorLevel._500);
+		LanguageLabel languageLabel = new LanguageLabel(labelTextPath, labelFont);
+		languageLabel.setStyle(labelStyle);
+		languageLabel.setWrap(true);
+		return languageLabel;
+	}
+
+	@Override
+	public void onResolutionChange() {
```

Comment:
nie jest spoko nadpisywanie tej metody, jak już to chociaż użyć super albo ustawić tą fukcję za pomocą resolutionChangeHandler()